test/rng_testsuite/statistics_tests/bernoulli_distribution_test.pass.cpp

```diff
@@ -0,0 +1,429 @@
+// -*- C++ -*-
+//===-- bernoulli_distribution_test.cpp ------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Test of bernoulli_distribution - check statistical properties of the distribution
+
+#include "support/utils.h"
+#include <iostream>
+
+#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
+#    include <CL/sycl.hpp>
+#    include <limits>
+#    include <oneapi/dpl/random>
+#    include <math.h>
+
+// Engine parameters
+constexpr auto a = 40014u;
+constexpr auto c = 200u;
+constexpr auto m = 2147483563u;
+constexpr auto seed = 777;
+
+int
+statistics_check(int nsamples, double p, bool* samples)
+{
+    // theoretical moments
+    double tM = p;
+    double tD = p * (1 - p);
+    double tQ = p;
+
+    // sample moments
+    double sum = 0.0;
+    double sum2 = 0.0;
+    for (int i = 0; i < nsamples; i++)
+    {
+        sum += samples[i];
+        sum2 += samples[i] * samples[i];
+    }
+    double sM = sum / nsamples;
+    double sD = sum2 / nsamples - sM * sM;
+
+    // comparison of theoretical and sample moments
+    double tD2 = tD * tD;
+    double s = ((tQ - tD2) / nsamples) - (2 * (tQ - 2.0 * tD2) / (nsamples * nsamples)) +
+               ((tQ - 3.0 * tD2) / (nsamples * nsamples * nsamples));
+
+    double DeltaM = (tM - sM) / sqrt(tD / nsamples);
+    double DeltaD = (tD - sD) / sqrt(s);
+
+    if (fabs(DeltaM) > 3.0 || fabs(DeltaD) > 3.0)
+    {
+        std::cout << "Error: sample moments (mean= " << sM << ", variance= " << sD
+                  << ") disagree with theory (mean=" << tM << ", variance= " << tD << ")";
+        return 1;
+    }
+
+    return 0;
+}
+
+template <class BoolType, class UIntType>
+int
+test(double p, int nsamples)
+{
+
+    sycl::queue queue;
+
+    // memory allocation
+    bool samples[nsamples];
+
+    constexpr int num_elems = oneapi::dpl::internal::type_traits_t<BoolType>::num_elems == 0
+                                  ? 1
+                                  : oneapi::dpl::internal::type_traits_t<BoolType>::num_elems;
+
+    // dpstd generation
+    {
+        sycl::buffer<bool, 1> buffer(samples, nsamples);
+
+        queue.submit([&](sycl::handler& cgh) {
+            auto acc = buffer.template get_access<sycl::access::mode::write>(cgh);

```

Comment:
can we use sycl 2020 way to create an accessor?

`sycl::accessor acc(buffer, cgh, sycl::write_only);`

Reply:
done